tensorflow/contrib/timeseries/python/timeseries/head.py

```diff
@@ -0,0 +1,344 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import re
+
+from tensorflow.contrib.framework.python.ops import variables
+from tensorflow.contrib.layers.python.layers import optimizers
+
+from tensorflow.contrib.timeseries.python.timeseries import feature_keys
+
+from tensorflow.python.estimator import estimator_lib
+from tensorflow.python.estimator.canned import head as head_lib
+from tensorflow.python.estimator.export import export_lib
+from tensorflow.python.framework import dtypes
+from tensorflow.python.framework import ops
+from tensorflow.python.ops import array_ops
+from tensorflow.python.ops import control_flow_ops
+from tensorflow.python.ops import math_ops
+from tensorflow.python.ops import state_ops
+from tensorflow.python.ops import variable_scope
+from tensorflow.python.util import nest
+
+
+def time_series_regression_head(
+        model, state_manager, optimizer, input_statistics_generator=None):
+  """Creates a `_Head` for regression using the mean squared loss.
+
+  Args:
+    weight_column: A string or a `_NumericColumn` created by
+      `tf.feature_column.numeric_column` defining feature column representing
+      weights. It is used to down weight or boost examples during training. It
+      will be multiplied by the loss of the example.
+    label_dimension: Number of regression labels per example. This is the size
+      of the last dimension of the labels `Tensor` (typically, this has shape
+      `[batch_size, label_dimension]`).
+
+  Returns:
+    An instance of `_Head` for linear regression.
+  """
+  return _TimeSeriesRegressionHead(
+    model, state_manager, optimizer, input_statistics_generator)
+
+
+class _TimeSeriesRegressionHead(head_lib._Head):  # pylint:disable=protected-access
+  """Returns a model function suitable for use with a tf.estimator.
+
+  Args:
+    model: The object (inheriting from Model) to create a function for.
+    state_manager: A state manager to wrap the model with (or
+        PassthroughStateManager if no state needs to be managed).
+    optimizer: An instance of `tf.train.Optimizer` to use for training.
+    input_statistics_generator: An InputStatisticsFromMiniBatch object from
+        math_utils.py, used for collecting statistics about input data during
+        training.
+  Returns:
+    The model function, suitable for passing to a tf.estimator.Estimator.
```

Comment:
Move the "Args:" to \_\_init__. The docstring and "Returns:" don't make much sense either place; it would be nice to write a one-line docstring for each.